include/plugin_interface/SC_SndBuf.h

```diff
@@ -39,9 +45,17 @@ class rw_spinlock {
     static const uint32_t locked_state = 0x80000000;
     static const uint32_t reader_mask = 0x7fffffff;
 
-#    ifdef __SSE2__
+/* this is a simplified version of nova::detail::pause() */
+#    if defined(_MSC_VER) /* Visual Studio Intel/ARM64 */
+    static inline void pause() { YieldProcessor(); }
+#    elif defined(__SSE2__) /* all modern Intel processors */
     static inline void pause() { _mm_pause(); }
+#    elif defined(__aarch64__) /* 64-bit ARM */
+    static inline void pause() { __asm__ __volatile__("isb"); }
+#    elif defined(__arm__) /* 32-bit ARM */
+    static inline void pause() { __asm__ __volatile__("yield"); }
 #    else
+#        warning "unknown architecture: fall back to busy-waiting"
     static inline void pause() {}
```

Comment:
Why not use the compiler GCC/clang builtin  `__builtin_ia32_pause`?

Reply:
Why should I? `_mm_pause` does the same thing and is the more portable of the two.

Reply:
Ah, okay, only asking because I'd seen people write spin-locks with `__builtin_ia32_pause` and not `_mm_pause`.

Looking again, Rust does exactly what your code does too.
https://doc.rust-lang.org/1.80.0/src/core/hint.rs.html#217